src/battle_script_commands.c

```diff
@@ -8980,6 +8981,57 @@ static void Cmd_various(void)
         else
             gBattlescriptCurrInstr += 9;
         return;
+    case VARIOUS_PHOTON_GEYSER_CHECK:
+    {
+        u32 attStat = gBattleMons[gActiveBattler].attack;
+        u8 atkStage = gBattleMons[gActiveBattler].statStages[STAT_ATK];
+        u32 spaStat = gBattleMons[gActiveBattler].spAttack;
+
+        attStat *= gStatStageRatios[atkStage][0];
+        attStat /= gStatStageRatios[atkStage][1];
+
+        atkStage = gBattleMons[gActiveBattler].statStages[STAT_SPATK];
+        spaStat *= gStatStageRatios[atkStage][0];
+        spaStat /= gStatStageRatios[atkStage][1];
+
+        if (attStat > spaStat)
+            gSwapDamageCategory = TRUE;
+        break;
+    }
+    case VARIOUS_SHELL_SIDE_ARM_CHECK: // 0% chance GameFreak actually checks this way according to DaWobblefet, but this is the only functional explanation at the moment
+    {
+        u32 attStat = gBattleMons[gBattlerAttacker].attack;
+        u8 atkStage = gBattleMons[gBattlerAttacker].statStages[STAT_ATK];
+        u32 attStatDef = gBattleMons[gBattlerTarget].attack;
```

Comment:
This is checking the wrong stats - should be the target's defenses, not its attack stats